File: kotlin/Benchmark.kt
import kotlin.system.measureTimeMillis

fun isPrime(n: Int): Boolean {
    if (n <= 1) return false
    for (i in 2..Math.sqrt(n.toDouble()).toInt()) {
        if (n % i == 0) return false
    }
    return true
}

fun fibonacci(n: Int): Int {
    if (n <= 1) return n
    return fibonacci(n - 1) + fibonacci(n - 2)
}

fun main() {
    val startMemory = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()

    var primes: List<Int> = listOf()
    var fib: Int = 0

    val elapsedTime = measureTimeMillis {
        primes = (2..100000).filter { isPrime(it) }
        fib = fibonacci(30)
    }

    val endMemory = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
    val memoryUsage = (endMemory - startMemory) / (1024 * 1024) // in MB

    println("Kotlin Benchmark")
    println("Time taken: $elapsedTime ms")
    println("Fibonacci: $fib")
    println("Primes: ${primes.size}")
    println("Memory Usage: $memoryUsage MB")
}

File: js/Benchmark.js
const os = require('node:os');
const startTime = Date.now();

function isPrime(n) {
    if (n <= 1) return false;
    for (let i = 2; i * i <= n; i++) {
        if (n % i === 0) return false;
    }
    return true;
}

function fibonacci(n) {
    if (n <= 1) return n;
    return fibonacci(n - 1) + fibonacci(n - 2);
}

function benchmark() {
    const initialMemory = process.memoryUsage().heapUsed / (1024 * 1024); // MB
    const initialCpuUsage = process.cpuUsage(); // Record CPU usage at the start

    const primes = [];
    for (let i = 2; i < 100000; i++) {
        if (isPrime(i)) primes.push(i);
    }

    const fib = fibonacci(30); // Adding Fibonacci calculation for extra load

    const elapsedTime = Date.now() - startTime;
    const finalMemory = process.memoryUsage().heapUsed / (1024 * 1024); // MB
    const cpuUsage = process.cpuUsage(initialCpuUsage); // Get CPU usage difference

    console.log("JavaScript Benchmark");
    console.log(`Time taken: ${elapsedTime} ms`);
    console.log(`Fibonacci number: ${fib}`);
    console.log(`Prime count: ${primes.length}`);
    console.log(`CPU Usage: ${cpuUsage.system} %`);
    console.log(`Memory Usage: ${(finalMemory - initialMemory).toFixed(2)} MB`);
}

benchmark();
